AQA-Advanced/Classes_hw/Book.js

```diff
@@ -0,0 +1,76 @@
+// class Book {
+//     constructor(name, author, publishDate){
+//         this.name = name;
+//         this.author = author;
+//         this.publishDate = publishDate;
+//     }
+// title() {
+//   console.log(this.name);
+//   }
+// writer() {
+//  console.log(this.author);
+// }
+// age() {
+//     console.log(this.publishDate);
+// }
+// }
+
+// const myBook = new Book("Hobbit", "J.R.R. Tolkien", 1984,);
+// const myNewBook = new Book ("Lord of the Ring", "J.R.R.Tolkien", 1991, );
+// const myOldBook = new Book (" Chipolino ", "T.Gorobynchyck", 1978,);
+// console.log(" My favourite Book " + myBook.name + " was written by " + this.author + ` in ` + (this.publishDate) )
+// console.log (" My second favourite Book " + myNewBook.name + " was written by " + myNewBook.author + ` in ` + (myNewBook.publishDate))
+// console.log(" The most favourite book from my childhood is " + myOldBook.name + " was written by " + myOldBook.author + ` in ` + (myOldBook.publishDate))
+
+class Book1 {
+    constructor(name, author, publishDate){
+        this.name = name;
+        this.author = author;
+        this.publishDate = publishDate;
+    }
+    get Name () {
+        return this.name
+      }
+      set Name(value){
+        if (value.lengthOfTheName > 4){
```

Comment:
 I think in the task the condition to check the type of value. not check the length